src/repository/cert.rs

```diff
@@ -316,91 +321,101 @@ impl Cert {
         Ok(())
     }
 
-    pub fn inspect_ca(&self, strict: bool) -> Result<(), ValidationError> {
-        self.inspect_ca_at(strict, Time::now())
-    }
-
-    pub fn inspect_ca_at(
-        &self, strict: bool, now: Time
-    ) -> Result<(), ValidationError> {
-        self.inspect_basics(strict, now)?;
+    pub fn inspect_ca(&self, strict: bool) -> Result<(), InspectionError> {
+        self.inspect_basics(strict)?;
         self.inspect_ca_basics(strict)?;
         self.inspect_issued(strict)
     }
 
-    pub fn inspect_ee(&self, strict: bool) -> Result<(), ValidationError> {
-        self.inspect_ee_at(strict, Time::now())
-    }
-
-    pub fn inspect_ee_at(
-        &self, strict: bool, now: Time
-    ) -> Result<(), ValidationError> {
-        self.inspect_basics(strict, now)?;
+    pub fn inspect_ee(&self, strict: bool) -> Result<(), InspectionError> {
+        self.inspect_basics(strict)?;
         self.inspect_issued(strict)?;
 
         // 4.8.1. Basic Constraints: Must not be present.
         if self.basic_ca.is_some(){
-            xerr!(return Err(ValidationError))
+            return Err(InspectionError::new(
+                "Basic Contraints extension \
+                 not allowed in end entity certificate"
+            ))
         }
 
         // 4.8.4. Key Usage. Bits for CA or not CA have been checked during
         // parsing already.
         if self.key_usage != KeyUsage::Ee {
-            xerr!(return Err(ValidationError))
+            return Err(InspectionError::new(
+                "invalid Key Usage extension \
+                 for end entity certificate"
+            ))
         }
 
         // 4.8.8.  Subject Information Access. We need the signed object
         // but not the other ones.
-        if self.ca_repository.is_some() || self.rpki_manifest.is_some()
-            || self.signed_object.is_none()
-        {
-            xerr!(return Err(ValidationError))
+        if self.ca_repository.is_some() {
+            return Err(InspectionError::new(
+                "id-ad-caRepository SIA instance \
+                 not allowed in end entity certificate"
+            ))
+        }
+        if self.rpki_manifest.is_some() {
+            return Err(InspectionError::new(
+                "id-ad-rpkiManifest SIA instance \
+                 not allowed in end entity certificate"
+            ))
+        }
+        if self.signed_object.is_none() {
+            return Err(InspectionError::new(
+                "missing id-ad-signedObject SIA instance \
+                 in signed object end entity certificate"
```

Comment:
This only seems to test signed_object yet also says it is missing from the ee cert, is this correct?

Reply:
I have added doc attributes to all these methods to make their purpose more clear.